contracts/modules/royalty/policies/IpRoyaltyVault.sol

```diff
@@ -280,112 +180,107 @@ contract IpRoyaltyVault is IIpRoyaltyVault, ERC20SnapshotUpgradeable, Reentrancy
+        $.poolInfo[token] += amount;
 
-    /// @notice Indicates whether the claimer has claimed the revenue tokens at a given snapshot
-    /// @param snapshotId The snapshot id
-    /// @param claimer The address of the claimer
-    /// @param token The address of the revenue token
-    function isClaimedAtSnapshot(uint256 snapshotId, address claimer, address token) external view returns (bool) {
-        return _getIpRoyaltyVaultStorage().isClaimedAtSnapshot[snapshotId][claimer][token];
+        emit RevenueTokenAddedToVault(token, amount);
     }
 
-    /// @notice Returns list of revenue tokens in the vault
-    function tokens() external view returns (address[] memory) {
-        return (_getIpRoyaltyVaultStorage().tokens).values();
-    }
+    function _update(address from, address to, uint256 amount) internal override {
+        if (from == address(0)) {
+            super._update(from, to, amount);
+            return;
+        }
+        if (from == to) revert Errors.IpRoyaltyVault__SameFromToAddress(address(this), from);
+        // when transferring RoyaltyTokens (Vault) from a user to another user:
+        // 1. clear pending rewards of the (from) user
+        // 2. clear pending rewards of the (to) another user, if pending rewards of another user is not zero
+        // 3. update the rewardDebt of the (to) another user to accBalancePerShare * userAmount
+        // 4. update the rewardDebt of the (from) user to accBalancePerShare * userAmount
+        // 5. transfer the RoyaltyTokens (Vault) from the (from) user to the (to) another user
+        if (balanceOf(from) == 0) revert Errors.IpRoyaltyVault__ZeroBalance(address(this), from);
+        if (balanceOf(from) < amount) revert Errors.IpRoyaltyVault__InsufficientBalance(address(this), from, amount);
 
-    /// @notice A function to calculate the amount of revenue token claimable by a token holder at certain snapshot
-    /// @param account The address of the token holder
-    /// @param snapshotId The snapshot id
-    /// @param token The revenue token to claim
-    /// @return The amount of revenue token claimable
-    function _claimableRevenue(address account, uint256 snapshotId, address token) internal view returns (uint256) {
         IpRoyaltyVaultStorage storage $ = _getIpRoyaltyVaultStorage();
+        address[] memory tokenList = $.tokens.values();
+        uint256 totalSupply = totalSupply();
+        for (uint256 i = 0; i < tokenList.length; i++) {
+            address token = tokenList[i];
+            _clearPendingRewards(from, token);
+            _clearPendingRewards(to, token);
+            $.claimerInfo[token][to] = ($.poolInfo[token] * (balanceOf(to) + amount)) / totalSupply;
+            $.claimerInfo[token][from] = ($.poolInfo[token] * (balanceOf(from) - amount)) / totalSupply;
+        }
 
-        // if the ip is tagged, then the unclaimed royalties are unavailable until the dispute is resolved
-        if (DISPUTE_MODULE.isIpTagged($.ipId)) return 0;
```

Comment:
One of the requirements as penalty of a dispute is the inability to claim revenue tokens from the vault when an IP is tagged.

Before we had this line:
`if (DISPUTE_MODULE.isIpTagged($.ipId)) return 0;`

In the new version we want to achieve the same requirement. Now for where to place this - if on transfer - would say that if we may want to measure the gas cost to confirm if it's ok to add it there or if another solution is preferrable.